From 0fc064f834cb15ad716ea45b3726fc6ca26e74bb Mon Sep 17 00:00:00 2001
From: David Runge <dave@sleepmap.de>
Date: Tue, 7 Dec 2021 22:46:17 +0100
Subject: [PATCH] Assert date headers are printable chars

hyperkitty/management/commands/hyperkitty_import.py:
Assert that chars in date headers are in the set of printable chars.
---
 hyperkitty/management/commands/hyperkitty_import.py | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/hyperkitty/management/commands/hyperkitty_import.py b/hyperkitty/management/commands/hyperkitty_import.py
index 2a50e933..48dd248d 100644
--- a/hyperkitty/management/commands/hyperkitty_import.py
+++ b/hyperkitty/management/commands/hyperkitty_import.py
@@ -31,6 +31,7 @@ from datetime import datetime
 from email import message_from_bytes, policy
 from email.utils import make_msgid, unquote
 from math import floor
+from string import printable
 from traceback import print_exc
 
 from django.conf import settings
@@ -131,7 +132,9 @@ class DbImporter(object):
     def _get_date(self, message, header, report_name):
         try:
             date = message.get(header)
-        except (TypeError, ValueError) as e:
+            if date:
+                assert all(char in printable for char in date)
+        except (AssertionError, TypeError, ValueError) as e:
             if self.verbose:
                 self.stderr.write(
                     "Can't get {} header in message {}{}: {}.".format(
-- 
GitLab

